tools/serverpod_cli/test/analyzer/entities/entity_analyzer/relation_keyword_test.dart

```diff
@@ -498,13 +507,160 @@ fields:
     var definition = SerializableEntityAnalyzer.extractEntityDefinition(
       protocol,
     );
+    SerializableEntityAnalyzer.resolveEntityDependencies([definition!]);
 
     SerializableEntityAnalyzer.validateYamlDefinition(
       protocol.yaml,
       protocol.yamlSourceUri.path,
       collector,
       definition,
-      [definition!],
+      [definition],
+    );
+
+    expect(
+      collector.errors.length,
+      greaterThan(0),
+      reason: 'Expected an error',
+    );
+
+    expect(
+      collector.errors.first.message,
+      'The "parent" property must be defined on id fields.',
+    );
+  });
+
+  test('Given a class ', () {
+    var collector = CodeGenerationCollector();
+
+    var protocol1 = ProtocolSource(
+      '''
+class: Example
+table: example
+fields:
+  parent: ExampleParent?, relation
+''',
+      Uri(path: 'lib/src/protocol/example.yaml'),
+      [],
+    );
+
+    var protocol2 = ProtocolSource(
+      '''
+class: ExampleParent
+table: example_parent
+fields:
+  name: String
+''',
+      Uri(path: 'lib/src/protocol/example.yaml'),
+      [],
+    );
+
+    var definition1 = SerializableEntityAnalyzer.extractEntityDefinition(
+      protocol1,
+    );
+
+    var definition2 = SerializableEntityAnalyzer.extractEntityDefinition(
+      protocol2,
+    );
+
+    var entities = [definition1!, definition2!];
+    SerializableEntityAnalyzer.resolveEntityDependencies(entities);
+
+    SerializableEntityAnalyzer.validateYamlDefinition(
+      protocol1.yaml,
+      protocol1.yamlSourceUri.path,
+      collector,
+      definition1,
+      entities,
+    );
+
+    var classDefinition = definition1 as ClassDefinition;
+
+    expect(
+        classDefinition.findField('parentId')?.parentTable, 'example_parent');
+  });
+
+  test('Given a class ', () {
+    var collector = CodeGenerationCollector();
+
+    var protocol1 = ProtocolSource(
+      '''
+class: Example
+table: example
+fields:
+  parent: InvalidClass?, relation
+''',
+      Uri(path: 'lib/src/protocol/example.yaml'),
+      [],
+    );
+
+    var definition1 = SerializableEntityAnalyzer.extractEntityDefinition(
+      protocol1,
+    );
+    var entities = [definition1!];
+    SerializableEntityAnalyzer.resolveEntityDependencies(entities);
+
+    SerializableEntityAnalyzer.validateYamlDefinition(
+      protocol1.yaml,
+      protocol1.yamlSourceUri.path,
+      collector,
+      definition1,
+      entities,
+    );
+
+    expect(
+      collector.errors.length,
+      greaterThan(0),
+      reason: 'Expected an error',
+    );
+
+    expect(
+      collector.errors.first.message,
+      'The class "InvalidClass" was not found in any protocol.',
+    );
+  });
+
+
+  test('Given a class ', () {
```

Comment:
Suggestion: add proper test names fir these.

Seems like the test names are missing.

Reply:
fixed later on.